### ISSUE TITLE
build error "could not bind to an lvalue of type 'const char [41]'"

### ISSUE THREAD

**Issue by pureland** [bitshares/bitshares-core#1952]
**Instructions**
Please include a detailed Title above. Next, please complete the following sections below:
* Build Error
  ` Reference to type 'const std::string' (aka 'const basic_string<char, char_traits<char>, allocator<char> >') could not bind to an lvalue of type 'const char [41]'`
* Build Environment
    Xcode Version 10.1 (10B61)/ clang version 4.0.1 (tags/RELEASE_401/final)

* Console Logs (optional)
```
bitshares-core/libraries/fc/include/fc/variant.hpp:571:10: error: reference to type 'const std::string' (aka 'const basic_string<char, char_traits<char>,
      allocator<char> >') could not bind to an lvalue of type 'const char [41]'
         _FC_ASSERT( max_depth > 0, "Recursion depth exceeded!" );
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
**Build Environment**
Details about the build environment, including the relevant required libraries. Much of this information can be found in the `CMakeFiles/CMakeOutput.log`. 
 - Host OS:             macOS Mojave
 - Host Physical RAM    32GB
 - Source Branch/Tag:   master 
 - OpenSSL Version:     1.1.0g
 - Boost Version:       1.59
 - C++ Compiler:       clang version 4.0.1 (tags/RELEASE_401/final)

**Comment by pmconrad** [comment]
Thanks. Can you please check if the error also occurs in the latest testnet release?
Need to
```
git checkout test-3.3.1
git submodule update --init --recursive
make
```
inside your checked out core directory.

**Comment by pmconrad** [comment]
@pureland @jmjatlanta could you test if this change fixes your problem: https://github.com/bitshares/bitshares-fc/pull/152/commits/78ffb6dbd3c3696f5838d005e3f159b42805172c ?

**Comment by pmconrad** [comment]
Not that it matters much, but that's two runtime-concatenations of strings, whereas mine is just a string literal.

**Comment by pmconrad** [comment]
Fixed with #1949
